src/mlagility/api/execute_trt.py

```diff
@@ -115,6 +126,65 @@ def format_field(line: str):
         e.writelines(decoded_error)
 
 
+"""
+In the average_power_and_utilization function, we eliminate the values below
+the idle_threshold at the beginning and end of the list because we want to 
+exclude the idle periods before and after the actual workload execution. By
+doing this, we can get a more accurate representation of the average power 
+consumption and GPU utilization during the execution of the workload.
+We set the Idle threshold to 3% utilization based on heuristics.
+"""
+IDLE_THRESHOLD = 3
+
+
+def average_power_and_utilization(power_readings):
+
+    if not power_readings:
+        return 0, 0
+
+    # Remove readings below the threshold from the beginning of the list
+    while power_readings and power_readings[0][0] <= IDLE_THRESHOLD:
+        power_readings.pop(0)
+
+    # Remove readings below the threshold from the end of the list
+    while power_readings and power_readings[-1][0] <= IDLE_THRESHOLD:
+        power_readings.pop()
+
+    if not power_readings:
+        return 0, 0
+
+    total_power = sum(power_draw for _, power_draw in power_readings)
+    total_utilization = sum(utilization for utilization, _ in power_readings)
+
+    average_power = total_power / len(power_readings)
+    average_utilization = total_utilization / len(power_readings)
+
+    return average_power, average_utilization
+
+
+def get_gpu_power_and_utilization():
+    try:
+        query = "nvidia-smi --query-gpu=utilization.gpu,power.draw --format=csv,noheader,nounits"
+        output = subprocess.check_output(query.split())
+        output = output.decode("utf-8").strip()
+
+        utilization, power_draw = output.split(", ")
+        return int(utilization), float(power_draw)
+
+    except Exception as e:
+        print(f"Error: {e}")
+        sys.exit(1)
```

Comment:
Any reason you're using `exit` instead of `raise` here? Our policy has been to avoid using `exit`.